micro-rdk/src/common/data_manager.rs

```diff
@@ -79,22 +79,29 @@ where
 
     pub async fn run(&mut self) -> Result<(), DataManagerError> {
         let mut loop_counter: u64 = 0;
-        let min_interval_ms = self.min_interval_ms();
         loop {
+            self.run_inner(loop_counter)?;
+            loop_counter += 1;
             Timer::after(self.min_interval).await;
-            if (self.sync_interval_ms() / min_interval_ms) % loop_counter == 0 {
-                self.sync()?;
-            }
-            for interval in self.collection_intervals() {
-                if (interval / min_interval_ms) % loop_counter == 0 {
-                    self.collect_and_store_readings(interval)?;
-                }
+        }
+    }
+
+    fn run_inner(&mut self, loop_counter: u64) -> Result<(), DataManagerError> {
+        let min_interval_ms = self.min_interval_ms();
+        if (loop_counter % (self.sync_interval_ms() / min_interval_ms)) == 0 && (loop_counter != 0)
+        {
+            self.sync()?;
+        }
+        for interval in self.collection_intervals() {
+            if loop_counter % (interval / min_interval_ms) == 0 {
+                self.collect_and_store_readings(interval)?;
             }
-            loop_counter += 1;
         }
+        Ok(())
     }
 
     fn sync(&mut self) -> Result<(), DataManagerError> {
+        println!("nsync");
```

Comment:
Remove

Reply:
Sorry, this was stale from a previous look at this PR. I know you disbanded nsync already.